### PR TITLE
Generate ice doc

### PR DESCRIPTION
**Pull request by jburel**
Minimum set of changes required to build the slice doc
Generation of the doc is tested by travis 
The commit 31035d8 will be removed when https://github.com/ome/ice-builder-gradle/pull/9 is merged and the repository tagged

In travis
* Check that the icedoc tasks are run
* Check that zip is created

Test locally:
* run ``gradle clean publishToMavenLocal`` (add ``-x javadoc`` to be quicker)

### CODE REVIEW

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
--exclude

File: build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ plugins {
     id "org.openmicroscopy.dsl" version "5.5.0-m6"
     id "org.openmicroscopy.blitz" version "5.5.0-m5"
     id "org.openmicroscopy.project" version "5.5.0-m5"
-    id "org.openmicroscopy.gradle.ice-builder.slice" version "1.5.0-m4"
+    id "org.openmicroscopy.gradle.ice-builder.slice" version "1.5.0-SNAPSHOT"
```
Comment (joshmoore) [inline build.gradle]: This should be achieved by merge-ci, right?

File: .travis.yml [hunk 2]
```diff
@@ -16,13 +16,18 @@ before_install:
...
-  - sudo apt-get install libzeroc-ice-java -y
+  - sudo apt-get install libzeroc-ice-java zeroc-ice-utils -y
   - sudo pip install https://github.com/ome/zeroc-ice-py-ubuntu1404/releases/download/0.1.0/zeroc_ice-3.6.4-cp27-none-linux_x86_64.whl
   - sudo ln -s /usr/local/bin/slice2py /usr/bin/slice2py
 
 install:
   - DIR=$PWD; (cd /tmp; gradle wrapper --gradle-version=5.2.1; mv .gradle gradle gradlew $DIR)
 
 script:
+  - git clone -b slice2html-mini --single-branch git://github.com/jburel/ice-builder-gradle
```
Comment (mtbc) [inline .travis.yml]: Using your fork is just temporary for review?

File: build.gradle (hunk 1 above)
Comment (jburel) [reply build.gradle]: yes this was necessary to pass travis

File: .travis.yml (hunk 2 above)
Comment (jburel) [reply .travis.yml]: yes. as indicated in the description this will be removed when ice-build-gradle changes are in and the repo is tagged

**Comment by mtbc** [comment]
Also to test that Insight on top of this still runs.

**Comment by jburel** [comment]
@mtbc This is done with merge-ci. See https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-insight-build/
insight was not building

File: publish.gradle [hunk 3]
```diff
@@ -4,10 +4,18 @@ task pythonZip(type: Zip) {
...
+    archiveClassifier.set("icedoc")
+    from compileIcedoc
+}
+
 publishing {
     publications {
         blitzPython(MavenPublication) {
             artifact pythonZip
         }
+        serverIcedoc(MavenPublication) {
```
Comment (mtbc) [inline publish.gradle]: Ice is still more a Blitz than server thing?

**Comment by jburel** [comment]
@mtbc the task has been renamed and travis is green i.e. the zip was created and tested via travis

**Comment by mtbc** [comment]
Looking good locally. Small RFE / idea: Should each of the zip files contain everything in a single parent folder?

**Comment by mtbc** [comment]
Passes functional testing.

**Comment by jburel** [comment]
@mtbc I think the structure is the same as the one in 5.4

**Comment by jburel** [comment]
The tmp commit has now been removed and it now uses the ice-builder tag.
Travis is still green.

**Comment by jburel** [comment]
Merging so we can rebase #51

**Pull request merged**